flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/ShardConsumer.java

```diff
@@ -155,6 +163,15 @@ public void run() {
 					// we can close this consumer thread once we've reached the end of the subscribed shard
 					break;
 				} else {
+					if (fetchIntervalMillis != 0) {
+						if (LOG.isDebugEnabled()) {
+							LOG.debug(
+								"Consumer {} of subtask {} is sleeping for {} milliseconds before fetching the next batch of records ...",
```

Comment:
I wonder if this log statement is really necessary. It can lead to quite a lot of log entries just for sleeping. (+ there is this other DEBUG log entry on each fetch)